core/src/main/java/io/grpc/internal/SpiffeIdParser.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Splitter;
+import java.util.Locale;
+
+/**
+ * Parses URI to SPIFFE ID if it matches SPIFFE ID standard.
+ * <p>
+ * @see <a href="https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md">standard</a>
+ */
+public final class SpiffeIdParser {
+
+  private static final String PREFIX = "spiffe://";
+
+  private SpiffeIdParser() {}
+
+  /**
+   * Parses a URI string, applies validation rules described in SPIFFE standard, and, in case of
+   * success, returns {@link SpiffeId} containing Trust Domain and Path of SPIFFE ID.
+   *
+   * @param uri a String representing a SPIFFE ID
+   * @return {@link SpiffeId} containing Trust Domain and Path of SPIFFE ID
```

Comment:
nit: Same here RE removing "containing..."

Reply:
Imho it's easier to read the current version, let's check with Eric

Reply:
Ok, after checking surrounding classes I realized that `@return` tag is almost never used. I changed it to similar to https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/AtomicBackoff.java#L43 style